### PR TITLE
Week 3 assessment

### PR DESCRIPTION
**Pull request by jeffreysjoshua**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -15,15 +15,56 @@
 
 // a) Create a test with expect statements for each of the variables provided.
 
-// Example input: 6
+var num1 = 6
 // Expected output: [1, 1, 2, 3, 5, 8]
 
-// Example input: 10
+var num2 = 10
```
Comment (sjproctor) [inline code-challenges.test.js:21]: It is best practice to put your test variables inside the scope of the test.

File: code-challenges.test.js [hunk 2]
```diff
@@ -15,15 +15,56 @@
...
+
+// create array to return. Set first and second index to 1 and 1
+var fib = [1, 1];
+// create if statement for edge cases where number is less than 3
+if (num < 3){
+  return [];
+// create else statement for other cases
+} else {
+// create for loop to iterate once per value of number
+for (var i = 2; i < num; i++) {
```
Comment (sjproctor) [inline code-challenges.test.js:48]: Typically we use `let` for the variable inside the for loop.

File: code-challenges.test.js [hunk 3]
```diff
@@ -15,15 +15,56 @@
...
+
+  // Fibonacci sequence formula: Xn = Xn-1 + Xn-2
+  let first = fib[i - 2];
+  let second = fib[i - 1];
+  fib[i] = first + second;
+// push the next value into the array
```
Comment (sjproctor) [inline code-challenges.test.js:54]: I'm wondering if the issue is here on line 53. I'm curious is redefining `fib[i]` is causing issues.

File: code-challenges.test.js [hunk 4]
```diff
@@ -15,15 +15,56 @@
...
+// I can't figure out why but this functions is pushing two of the last index
+// into the array. Using .pop() gives me the desired output. Not how I wanted
+// to solve it but it works.
+  fib.pop()
+  return fib
+}
+}
+
+// Note: function produces green "passed" message. Function refactored, as needed.
+
```
Comment (sjproctor) [inline code-challenges.test.js:67]: Nice work on grinding that one.

File: code-challenges.test.js [hunk 5]
```diff
@@ -52,5 +123,35 @@ var numbersToAdd2 = [0, 7, -8, 12]
...
+// create new array variable and set index zero to what it already is
+  var summed = [array[0]];
+// iterate over the array starting with index 1
+  for (var i=1; i < array.length; i++){
+// push the value of array[i], with the previous value, in the new array, into the new array
+    summed.push(array[i] + summed[i -1])
+  }
+  }
+  return summed
+}
```
Comment (sjproctor) [inline code-challenges.test.js:157]: Nice work! I think a refactor with some higher-order functions could shorten your code.

File: interview-questions.md [hunk 6]
```diff
@@ -9,59 +9,59 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is `this` in JavaScript?
 
-  Your answer:
+  Your answer: "This" in JavaScript is a way to refer to a key within an object. You can place "this" before a ".(key name)", and use it to manipulate the value of the key within the object, or to call state/setstate in a react app.
```
Comment (sjproctor) [inline interview-questions.md:12]: Nice!

File: interview-questions.md [hunk 7]
```diff
@@ -9,59 +9,59 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: Yarn is a command used to create and open a react app. It imports the needed dependencies to create a "shell" react app. When yarn is ran, all files saved on the React application are ran, and any changes made since the previous save are changed in the application.
 
-  Researched answer:
+  Researched answer: I found that yarn is a package manager that was created by FB as an alternative to the npm package manager. It allows you to still access npm dependencies but is more efficient.
```
Comment (sjproctor) [inline interview-questions.md:46]: Yes, yarn will connect dependencies. The yarn.lock, package.json, and node_nodules will be updated.

File: interview-questions.md [hunk 8]
```diff
@@ -9,59 +9,59 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 6. STRETCH: What is an anonymous function in JavaScript?
 
-  Your answer:
+  Your answer: I really don't know, but if I had to make an educated guess, I'd say that an "anonymous function" is a function that does not have a definition- which is what would make it anonymous..
 
-  Researched answer:
+  Researched answer: Ah my researched answer is pretty close! I also found that an AF cannot be used after its creation because it doesn't have a name and therefor cannot be called. However, you can assign an AF to a variable and use that variable later.
```
Comment (sjproctor) [inline interview-questions.md:54]: Nice! Really great jobs on the interview questions!

File: react-debugging.html [hunk 9]
```diff
@@ -62,26 +62,32 @@
...
+                     {card}
+                   </div>
+                 )
+               })}
+             </div>
+             </>
+         )
+       }
+     }
+
```
Comment (sjproctor) [inline react-debugging.html:90]: Boom!

**Review by sjproctor — APPROVED** [review]
Great work Josh! Way to work through code challenges. It looks like you are using some older JavaScript code syntax in your challenges, particularly the variable declarations. Most of the time that syntax will work but it is best practices to use more updated concepts.
